File: Python/7. Reverse Integer.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        flag = 1
        res = 0

        if x < 0:
            x = -x
            flag = -1

        while True:
            if (x <= 0):
                break
            res = int(res * 10 + x % 10)
            x = int(x / 10)

        res = flag * res

        if res < -2 ** 31 or res > 2 ** 31 - 1:
            return 0
        
        return res

File: C++/7. Reverse Integer.cpp

class Solution {
public:
    int reverse(int x) {
        long flag = 1;
        long res = 0;

        if (x < 0) {
            x = -x;
            flag = -1;
        }


        while (x) {
            res = res * 10 + x % 10;
            x /= 10;
        }

        res = flag * res;
        
        if (res < -pow(2, 31) || res > pow(2, 31) - 1)
            return 0;
        else
            return res;

    }
};